### PR TITLE
Add metadata management context

### PR DESCRIPTION
**Pull request by adimiz1**
### Brief Summary of Changes
Add metadata management context

#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [ ] Refactoring
- [x] New feature
- [ ] Bug fix
- [x] Adds more tests

#### Are tests included?
- [x] Yes
- [ ] No

#### Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

File: api/lib/uploader/uploader.dart [hunk 1]
```diff
@@ -51,4 +52,43 @@ class Uploader {
...
+
+  Future<UploaderResponse<UploadResult>> removeAllContext(RemoveAllContextParams params) {
+    params.command = ContextCommand.remove_all.name;
+    UploaderRequest request = UploaderRequest(params);
+    return _uploaderUtils.callApi(request, 'context', options: SharedParams());
+  }
+
+  Future<UploaderResponse<UploadResult>> updateMetadata(DestroyParams params) {
+    UploaderRequest request = UploaderRequest(params);
+    return _uploaderUtils.callApi(request, 'destroy', options: SharedParams());
```
Comment (const-cloudinary) [inline api/lib/uploader/uploader.dart:72]: what is `destroy`? and why all methods below have it?

File: api/lib/src/request/model/uploader_params.dart [hunk 2]
```diff
@@ -366,15 +405,31 @@ class UploadParams extends UploaderParams {
     if (context == null) {
       return null;
     }
-    return context.entries.map((map) {
-      if (map.value is List<dynamic>) {
-        _encodeList(map.value);
-      } else if (map.value is String) {
-        _cldEncodeSingleContextItem((map.value as String));
```
Comment (const-cloudinary) [inline api/lib/src/request/model/uploader_params.dart:373]: What happened to this function: `_cldEncodeSingleContextItem` is it still in use elsewhere?

File: api/lib/src/request/model/uploader_params.dart [hunk 3]
```diff
@@ -83,6 +83,45 @@ class DestroyParams extends UploaderParams {
   }
 }
 
+class ContextParams extends UploaderParams {
```
Comment (const-cloudinary) [inline api/lib/src/request/model/uploader_params.dart:86]: Do we need to extend `UploaderParams`? looks like most of them are irrelevant for this operation.

**Review by const-cloudinary — CHANGES_REQUESTED** [review]

File: api/lib/src/request/model/uploader_params.dart (hunk 3 above)
Comment (adimiz1) [reply api/lib/src/request/model/uploader_params.dart:86]: Fixed to take the layer above that

**Review by const-cloudinary — APPROVED** [review]

**Pull request merged**